.github/workflows/ci.yaml

```diff
@@ -142,25 +142,42 @@ jobs:
     if: github.event_name == 'pull_request'
     steps:
       - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            linter:
+            - 'Cargo.toml'
+            - 'Cargo.lock'
+            - 'crates/**'
+            - '!crates/ruff_python_formatter/**'
+            - '!crates/ruff_formatter/**'
+            - '!crates/ruff_dev/**'
+            - '!crates/ruff_shrinking/**'
+
       - uses: actions/setup-python@v4
+        if: steps.filter.outputs.linter == 'true'
         with:
           python-version: ${{ env.PYTHON_VERSION }}
 
       - uses: actions/download-artifact@v3
+        if: steps.filter.outputs.linter == 'true'
```

Comment:
This is silly but GitHub doesn't support early returns. The *proposed* alternative is to have another job but that feels bad in its own way because it pollutes the checks panel with even more output